plugins/jobs/src/js/pages/JobDetailPage.js

```diff
@@ -0,0 +1,195 @@
+import JobRunHistoryTable from "./JobRunHistoryTable";
+
+class JobDetailPage extends mixin(TabsMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.tabs_tabs = {
+      runHistory: "Run History",
+      configuration: "Configuration"
+    };
+
+    this.state = {
+      currentTab: Object.keys(this.tabs_tabs).shift()
+    };
+  }
+
+  getNavigationTabs() {
+    return <ul className="menu-tabbed">{this.tabs_getUnroutedTabs()}</ul>;
+  }
+
+  getDestroyConfirmDialog() {
+    return null;
+  }
+
+  getPrettySchedule(job) {
+    const schedules = job.schedules;
+    if (schedules == null || schedules.length === 0) {
+      return null;
+    }
+
+    const schedule = schedules[0];
+    if (schedule.enabled) {
+      return prettycron.toString(schedule.cron);
+    }
+  }
+
+  getSubTitle(job) {
+    const nodes = [];
+    const scheduleText = this.getPrettySchedule(job);
+    let longestRunningTask = null;
+    const longestRunningActiveRun = job
+      .getActiveRuns()
+      .getLongestRunningActiveRun();
+
+    if (longestRunningActiveRun != null) {
+      longestRunningTask = longestRunningActiveRun
+        .getTasks()
+        .getLongestRunningTask();
+    }
+
+    if (longestRunningTask == null && scheduleText == null) {
+      return null;
+    }
+
+    if (scheduleText != null) {
+      nodes.push(
+        <p className="text-overflow flush" key="schedule-text">
+          <Icon
+            className="icon-margin-right"
+            key="schedule-icon"
+            color="grey"
+            id="repeat"
+            size="mini"
+          />
+          <span>Scheduled {StringUtil.lowercase(scheduleText)}</span>
+        </p>
+      );
+    }
+
+    if (longestRunningTask != null) {
+      const dateCompleted = longestRunningTask.getDateCompleted();
+
+      const status = TaskStates[longestRunningTask.getStatus()];
+      const statusClasses = classNames("job-details-header-status", {
+        "text-success":
+          status.stateTypes.includes("success") &&
+          !status.stateTypes.includes("failure"),
+        "text-danger": status.stateTypes.includes("failure")
+      });
+
+      let timePrefix = null;
+      const shouldSuppressRelativeTime = dateCompleted == null;
+
+      if (shouldSuppressRelativeTime) {
+        timePrefix = "for ";
+      }
+
+      nodes.push(
+        <span className={statusClasses} key="status-text">
+          {status.displayName}
+        </span>,
+        <span key="time-running">
+          <TimeAgo
+            prefix={timePrefix}
+            suppressSuffix={shouldSuppressRelativeTime}
+            time={longestRunningActiveRun.getDateCreated()}
+          />
+        </span>
+      );
+    }
+
+    return nodes;
+  }
+
+  renderConfigurationTabView(job) {
+    return <JobConfiguration job={job} />;
+  }
+
+  renderRunHistoryTabView(job) {
+    return <JobRunHistoryTable job={job} />;
+  }
+
+  getActions() {
+    const actions = [];
+
+    actions.push({
```

Comment:
Are we going to add more actions depending on any logic? Curious why don't we just return an array directly here?

Reply:
Yes, indeed. But that shouldn't hinder us for now, I will refactor this 👍

Reply:
Please do not touch or care about this code. I already took care of it while doing the menus.